prov/efa/src/efa_device.c

```diff
@@ -97,6 +97,15 @@ int efa_device_construct(struct efa_device *efa_device,
 		goto err_close;
 	}
 
+	if (-FI_ENODATA == fi_param_get_int(&rxr_prov, "use_device_rdma", &tmp_value)) {
+		if (efa_device->ibv_attr.vendor_part_id < 0xefa2) {
```

Comment:
Don't do this.  Explicitly compare against 0xefa0 and 0xefa1.  There's no guarantee that the device id will always start with 0xefa?.  We one day will have more than 16 EFA versions, and if for some reason, we decremented one of the first characters (like made the one after 0xefaf be 0xef00), then this would not do the right thing.

Reply:
Could you pull out a function:
```c
bool canUseDeviceRdma(int vendor_part);
```

Reply:
Thank you! Fixed in the latest revision.

Reply:
>Could you pull out a function:
bool canUseDeviceRdma(int vendor_part);

This is only a simple `if` and I am not sure introducing an extra function can be better.